### ISSUE TITLE
Parsing should stop when END is reached

### ISSUE THREAD

**Issue by thiloplanz** [danielrendall/Metaphor#3]
If the InputStream contains data beyond the MTEF data, the parser will continue to try to read it.
It should probably terminate when it has reached the END tag.

**Comment by danielrendall** [comment]
Yes, it probably should :-). At the moment, I'm assuming that it will get a valid MTEF stream without additional cruft. Presumably, once we hit the initial PILE or LINE record after the initial font stuff, we should keep track of how deep we are in nested records and stop when we've hit the corresponding END (perhaps warn in some way if the stream contains extra data?)

**Comment by thiloplanz** [comment]
I think it is quite easy. In the while loop in MTEFParser#doParse, just look at the record just parsed, and if it is an instance of END, break out of the loop. That seems to have worked for me, I will test it a bit more next week, and send you a patch.

**Comment by danielrendall** [comment]
Sounds sensible, thanks for taking the time to look at this :-)
